### DISCUSSION TITLE
How to disable one of the I2C interface (flexcomm)

### DISCUSSION THREAD

**Discussion by MarsMSWu** [zephyrproject-rtos/zephyr#75950] [Q&A] [answered]
Hi all...

I use the LPC55s69 with FC0~FC7, some for UART and some for I2C with different speed.
If enter deep-sleep mode to reduce power consumption, is possible to disable some I2C channel ? (disable some flexcomm)
Not need disable all...

Thanks.
Mars

**Comment by jacob-wienecke-nxp** [comment, marked as answer]
Hi Mars,

Just to clarify, the mode supported is sleep, not deep-sleep.

You can disable the peripheral clocks when going in to sleep mode and reenable them when waking up. This will reduce the current draw.

I recommend  taking a look at the [Power Management APIs](https://docs.zephyrproject.org/latest/services/pm/api/index.html#c.pm_notifier_register). The [struct pm_notifier](https://docs.zephyrproject.org/latest/services/pm/api/index.html#c.pm_notifier_register) contains callbacks that are called when entering and exiting power states. This can aid in managing the peripheral clocks when entering/exiting deep-sleep mode.

Thanks, Jacob

**Reply by MarsMSWu** [reply]
Hi Jacob,
Do you mean I should configure a wakeup source with ISR and enter "pm_state_force( )" then "pm_system_resume( )" in wakeup ISR event ?

Thanks.
Mars

**Reply by MarsMSWu** [reply]
Hi Jacob,
Ispossible to dynamic change I2C speed from 400K -> 100k ??

Thanks.
Mars

**Reply by jacob-wienecke-nxp** [reply]
1) The System PM feature is triggered in the Idle thread.  When the app has nothing to do, it enters the idle thread, which can trigger the PM to enter Sleep mode. Before it enters Sleep mode, it will notify the callbacks already registered in the app.  **The app can then disable the Flexcomm clock.** Disabling the Flexcomm clock will achieve the desired functionality as described in your orignal question. After waking from Sleep mode, the notify callback is called again.  **The app can then re-enable the clocks**.  The Flexcomm will resume operation once the clock has been re-enabled.

2) "Changing I2C Speed Dynamically" Is unrelated to the discussion on "disabling one I2C interface" I request that you create a new discussion. As far as I am aware, it is not typical to change the baud rate past the initial configuration.

**Reply by MarsMSWu** [reply]
Hi Jacob,

I think maybe not need to configure CONFIG_PM=y in proj.conf because the system should default enter to k_cpu_idle while nothing to do...
If Zephyr enter to k_cpu_idle, it should have lower power consumption...Am I right ?
![image](https://github.com/user-attachments/assets/16418a75-ee16-4852-a601-51680a397504)


Thanks, Mars.

**Reply by jacob-wienecke-nxp** [reply]
Mars,

CONFIG_PM=y is required to enable the PM subsystem.

CONFIG_PM=y is not a requirement to enter SLEEP mode on this SOC. The ARM Cortex-M core will execute WFI, which will also enter SLEEP mode on this SOC. 

However, CONFIG_PM=y is required for the PM notifications. In order to leverage the PM notifications to manage peripheral clocks (disabling/re-enabling the clock to I2C) you will need CONFIG_PM=y.

Thanks, Jacob

**Reply by MarsMSWu** [reply]
Hi Jacob,

Thanks for your reply...
I will try to add CONFIG_PM=y and test again...confirm.

Thanks, Mars
